shared/validate.js

```diff
@@ -82,17 +82,26 @@ module.exports = {
     if (functions && Object.keys(functions).length !== 0) {
       functionNames = Object.keys(functions);
 
-      // Check that runtime is authorized
+      // Check that default runtime is authorized
       const extensions = RUNTIMES_EXTENSIONS[this.runtime];
       if (!extensions) {
         const availableRuntimesMessage = Object.keys(RUNTIMES_EXTENSIONS).join(', ');
         functionErrors.push(`Runtime ${this.runtime} is not supported. Function runtime must be one of the following: ${availableRuntimesMessage}`);
       }
 
       functionNames.forEach((functionName) => {
+
         const func = functions[functionName];
-        if (this.runtime === 'golang') { // Golang runtime does not work like other runtimes, we should ignore validation for it
-          return;
+        
+        // Check that function's runtime is authorized if existing
+        if (func.runtime && func.runtime !== 'golang'){
+          const extensions = RUNTIMES_EXTENSIONS[func.runtime];
+          if (!extensions) {
+            const availableRuntimesMessage = Object.keys(RUNTIMES_EXTENSIONS).join(', ');
+            functionErrors.push(`Runtime ${this.runtime} is not supported. Function runtime must be one of the following: ${availableRuntimesMessage}`);
+          }
+        } else if (func.runtime === 'golang' || this.runtime === 'golang') {
```

Comment:
you could check this before and remove the first if

Reply:
It won't work because if the global runtime is golang and the function's runtime is node10 it will enter in this condition and not going in the other one -> something that we do not want

Reply:
right !